The "matrix-mult" module
========================

        >>> matrix_mul([[1, 2, 3]], [4, 5, 6]], [7, 8, 9]],)
        Traceback (most recent call last):
            ...
        TypeError: matrix_mul() takes 2 positional arguments but 3 were given
        >>> matrix_mul([1, 2, 3], [4, 5, 6])
        Traceback (most recent call last):
            ...
        TypeError: m_a must be a list of lists
        >>> matrix_mul([[1, 2, 3]], [4, 5, 6]], [7, 8, 9]])
        Traceback (most recent call last):
            ...
        TypeError: m_b must be a list of lists
        >>> matrix_mul([[1, 2, 3]], [4, 5, 6]], [7, 8])
        Traceback (most recent call last):
            ...
        ValueError: m_a and m_b can't be multiplied
        >>> matrix_mul([[1, 2], [3, 4]], [5, 6], [7, 8])
        Traceback (most recent call last):
            ...
        ValueError: m_a and m_b can't be multiplied
        >>> matrix_mul([[1, 2, 3]], [4, 5, 6], [7, 8, 9]], [10, 11, 12], [13, 14, 15], [16, 17, 18])
        Traceback (most recent call last):
            ...
        ValueError: m_a and m_b can't be multiplied
        >>> matrix_mul([[1, 2, 3]], [4, 5, 6], [7, 8, 9]], [10, 11, 12])
        [[58, 64, 70]]
        >>> matrix_mul([[1, 2], [3, 4]], [5, 6, 7]])
        Traceback (most recent call last):
            ...
        TypeError: each row of m_b must be of the same size
        >>> matrix_mul([[1, 2], [3, 4]], [5, 6]])
        Traceback (most recent call last):
            ...
        ValueError: m_a and m_b can't be multiplied

